vscode-trace-webviews/src/trace-explorer/time-range/vscode-time-range-data-widget.tsx

```diff
@@ -0,0 +1,136 @@
+  static LABEL = 'Time Range Data';
+
+  constructor(props: TimeRangeDataWidgetProps) {
+      super(props);
+      this._signalHandler = new VsCodeMessageManager();
+      this._reactRef = React.createRef();
+      
+      window.addEventListener('message', event => {
+
+
+          const { command, data } = event.data;
+
+          if (command === 'restore-view') {
+            this.restoreState();
+            return;
+          }
+
+          const experiment = data.wrapper ? convertSignalExperiment(JSONBig.parse(data.wrapper)) : undefined;
+          const parsedData: TimeRangeUpdatePayload = !experiment ? JSONBig.parse(data) : undefined;
+
+          switch (command) {
+            case VSCODE_MESSAGES.EXPERIMENT_SELECTED:
+                signalManager().fireExperimentSelectedSignal(experiment);
+                break;
+            case VSCODE_MESSAGES.EXPERIMENT_UPDATED:
+                if (experiment) {
+                    signalManager().fireExperimentUpdatedSignal(experiment);
+                }
+                break;
+            case VSCODE_MESSAGES.EXPERIMENT_CLOSED:
+                if (experiment) {
+                    signalManager().fireExperimentClosedSignal(experiment);
+                }
+                break;
+            case VSCODE_MESSAGES.SELECTION_RANGE_UPDATED:
+                if (parsedData) {
+                    signalManager().fireSelectionRangeUpdated(parsedData);
+                }
+                break;
+            case VSCODE_MESSAGES.VIEW_RANGE_UPDATED:
+                if (parsedData) {
+                    signalManager().fireViewRangeUpdated(parsedData);
+                }
+                break;
+          }
+
+          this.storeState();
+
+      });
+  }
+
+
+  componentWillUnmount = (): void => {
+      signalManager().off(Signals.REQUEST_SELECTION_RANGE_CHANGE, this.onRequestSelectionChange);
+  }
+
+  onRequestSelectionChange = (payload: TimeRangeUpdatePayload): void => {
+      this._signalHandler.requestSelectionRangeChange(payload);
+  };
+
+  componentDidMount = (): void => {
+      this._signalHandler.notifyReady();
+      signalManager().on(Signals.REQUEST_SELECTION_RANGE_CHANGE, this.onRequestSelectionChange);
+
+  }
+
+  storeState = (): void => {
+      if (this._reactRef.current) {
+        const { state, experimentDataMap } = this._reactRef.current;
+        vscode.setState({
+            activeData: JSONBig.stringify(state.activeData),
+            experimentDataMap: JSONBig.stringify(Array.from(experimentDataMap))
+        });
+      }
+  }
+
+  restoreState = (): void => {
+      if (this._reactRef.current) {
+          const persistedData = vscode.getState();
+          const activeData = JSONBig.parse(persistedData.activeData);
+          const experimentDataMap: Map<string, ExperimentTimeRangeData> = new Map(JSONBig.parse(persistedData.experimentDataMap));
+          this._reactRef.current.restoreState(activeData, experimentDataMap);
```

Comment:
error TS2339: Property 'restoreState' does not exist on type 'ReactTimeRangeDataWidget'.

Reply:
This error is also linked to the theia-trace-extension changes.